lib/matplotlib/axes/_axes.py

```diff
@@ -2333,8 +2333,16 @@ def broken_barh(self, xranges, yrange, **kwargs):
         .. plot:: mpl_examples/pylab_examples/broken_barh.py
         """
         # process the unit information
-        self._process_unit_info(xdata=xranges[0],
-                                ydata=yrange[0],
+        if len(xranges):
+            xdata = xranges[0]
```

Comment:
Probably safer to do `xdata = next(iter(xrange))`  in a `try...except` block.  pandas `Series` slice on index rather than position

```python
try:
    xdata = next(iter(xranges))
except (StopIteration, TypeError):
    xdata = None
```